### DISCUSSION TITLE
Support for multiple locations

### DISCUSSION THREAD

**Discussion by jbchr** [open-meteo/open-meteo#247] [Q&A]
Hello hello, 

thanks for this nice project! Is there currently support (or anything planned) for querying forecast data for multiple locations at once without having to do one api call for each location I want to query? 

Thanks a lot!
Janek

**Comment by patrick-zippenfenig** [comment]
This has been frequently requested, but it is not yet available. There are 2 issues:
- The API output format needs to allow multiple locations at once. This include JSON, CSV and XLSX formats. Additionally, I would like to implement more efficient binary encodings like protobuf or flatbuffer.
- Server load is increased. Especially, for long time-series historical data queries

For which use case do you want to query multiple locations at once?

**Comment by jbchr** [comment]
Thanks for you quick reply Patrick, and sorry I searched for it but didn't find similar requests! 

I'm building a hobby project, where I want to show where on earth there is good weather in the next days (I only need weathercode, min, max temp for 7 days). For this i'm dividing all continents into hexagons that display the weather condition in color codes. These are approximately ~10.000 hexagons that I want to get weather info for. I was planing on updating this once a day to stay within the api limit but I get frequent timeouts or refused connections even with throttling the requests. This is why I was wondering if there is support to query multiple locations at once.

**Comment by phillipseamore** [comment]
This is not a correct way of doing it, I'd even call it abusive. It will be a lot faster and simpler for you to just get the GFS 0.5 from Nomads and query it out yourself.

**Comment by jbchr** [comment]
@phillipseamore I agree that this is not a sustainable solution. Up till now I only used it to generate the mock data to build the frontend. The GFS 0.5 from Nomads might be what I am looking for, thank you for pointing me into this direction!

**Reply by patrick-zippenfenig** [reply]
There might be even the 1° GFS still on the nomads server. For global weather maps, it is easier to just use GRIB files. Maybe at some point, I will also provide APIs to generate maps, but until now I only focused on time-series APIs based on high resolution models.
